src/SWIG_files/wrapper/TopoDS.i

```diff
@@ -1980,11 +1980,44 @@ Returns the type as a term of the shapeenum enum: vertex, edge, wire, face, ....
 ") ShapeType;
 		virtual TopAbs_ShapeEnum ShapeType();
 
+
+%extend{
+    bool __ne_wrapper__(const opencascade::handle<TopoDS_TShape> & other) {
+    if (self!=other) return true;
+    else return false;
+    }
+}
+%pythoncode {
+def __ne__(self, right):
+    try:
+        return self.__ne_wrapper__(right)
+    except:
+        return True
+}
+
+%extend{
+    bool __eq_wrapper__(const opencascade::handle<TopoDS_TShape> & other) {
+    if (self==other) return true;
+    else return false;
+    }
+}
+%pythoncode {
+def __eq__(self, right):
+    try:
+        return self.__eq_wrapper__(right)
+    except:
+        return False
+}
 };
 
 
 %make_alias(TopoDS_TShape)
 
+%extend TopoDS_TShape {
+    size_t __hash__() {
+    return opencascade::hash(self);}
+};
+
```

Comment:
The ```TopoDS_Shape``` inherits from ```Standard_Transient```, I think the ```__hash__``` method could be moved to the base class, and would also benefit to all other classes derived from ```Standard_Transient```

Reply:
Good point, I'll move the `__hash__` to `Standard_Transient` together with the `__eq__` and `__neq__` methods.

Reply:
no need to, I did it there's a build going on